File: app/src/main/java/rssdemo/andras/hu/rssdemo/ui/feed/FeedView.java
package rssdemo.andras.hu.rssdemo.ui.feed;


import com.facebook.share.model.ShareLinkContent;

import java.util.List;

import rssdemo.andras.hu.rssdemo.data.Subscription;

public interface FeedView {
    FeedAdapter getAdapter();

    void populateDrawerMenu(List<Subscription> subscriptions);

    void showError();

    void showLoaderOverlay();

    void hideLoaderOverlay();

    void setTitle(String name);

    void shareContent(ShareLinkContent content);
}

File: app/src/main/java/rssdemo/andras/hu/rssdemo/ui/feed/FeedView.kt
package rssdemo.andras.hu.rssdemo.ui.feed


import com.facebook.share.model.ShareLinkContent
import rssdemo.andras.hu.rssdemo.data.FeedItem

import rssdemo.andras.hu.rssdemo.data.Subscription

internal interface FeedView {

    fun setFeedItems(items: List<FeedItem>)

    fun populateDrawerMenu(subscriptions: List<Subscription>)

    fun showError()

    fun showLoaderOverlay()

    fun hideLoaderOverlay()

    fun setTitle(name: String)

    fun showFacebookShareDialog(content: ShareLinkContent)
}
